invenio_records_lom/utils/__init__.py

```diff
@@ -0,0 +1,439 @@
+        res = []
+        for subkey in dotted_key.split("."):
+            try:
+                res.append(int(subkey))
+            except ValueError:
+                res.append(subkey)
+        return res
+
+
+def get_oefosdict(language_code: str = "de") -> dict[str, str]:
+    """Get oefos-dict, which maps OEFOS-codes to subject-names."""
+    oefosdict = {}  # to be result
+
+    filenames_by_language = {
+        "de": "OEFOS2012_DE_CTI_20211111_154218_utf8.csv",
+        "en": "OEFOS2012_EN_CTI_20211111_154228_utf8.csv",
+    }
+    if language_code.lower() not in filenames_by_language:
+        raise ValueError(f"OEFOS aren't available for language_code {language_code!r}.")
+    filename = filenames_by_language[language_code.lower()]
+
+    oefos_path = Path(__file__).resolve().parent / filename
+
+    line_iter = iter(oefos_path.read_text().splitlines())
+    __ = next(line_iter)  # discard header
+    for line in line_iter:
+        # line is of form `{depth-level};"{EDV-code}";"{code}";"{title}";"{short title}"`
+        # e.g. `4;"603125";"603125";"Metaethik";""`
+        __, edv_code, __, name, __ = line.split(";")
+        oefosdict[edv_code.strip('"')] = name.strip('"')
+    return oefosdict
+
+
+def langstringify(text: str, lang: t.Optional[str] = "x-none") -> dict:
+    """Wraps `text` and `lang` into a langstring-dict as per LOM-standard.
+
+    If `lang` is falsy, the output won't have a "lang"-field.
+    """
+    langstring = {}
+    langstring["#text"] = text
+    if lang:
+        langstring["lang"] = lang
+    return {"langstring": langstring}
+
+
+def vocabularify(value: str, source: str = "LOMv1.0") -> dict:
+    """Wraps `value` and `source` into a vocabulary-dict as per LOM-standard."""
+    return {
+        "source": langstringify(source),
+        "value": langstringify(value),
+    }
+
+
+def catalogify(value: str, catalog: str) -> dict:
+    """Wraps `value` and `catalog` into a catalog-dict as per LOM-standard."""
+    return {
+        "catalog": catalog,
+        "entry": langstringify(value),
+    }
+
+
+def durationify(datetime: str, description: str):
+    """Wraps `datetime` and `description` into a duration-dict as per LOM-standard.
+
+    If either parameter is falsy, the output won't have the corresponding field.
+    """
+    if not datetime and not description:
+        raise ValueError("At least one of `datetime`, `description` need be truthy.")
+
+    inner = {}
+    if datetime:
+        inner["datetime"] = datetime
+    if description:
+        inner["decription"] = description
+
+    return {"duration": inner}
+
+
+class LOMMetadata:
+    """Helper-class for easy creation of LOM-metadata JSONs."""
+
+    # learningresourcetypes according to `https://w3id.org/kim/hcrt/scheme`
+    # used in LOM.educational.learningresourcetype
+    # this maps (url-ending -> label-dict) where label-dict maps (lang-code -> label)
+    learningresourcetype_labels = {
```

Comment:
please move json into a json file. in my opinion this should not be hard coded into python code